0011-command-execution-protocol/README.md

```diff
@@ -0,0 +1,170 @@
+---
+name: Command Execution Protocol
+shortname: 11/CEP
+status: raw
+editor: Alexey Shmalko <ashmalko@kaaiot.io>
+---
+
+<!-- toc -->
+
+# Introduction
+Command Execution Protocol is an endpoint-aware 1/KP extension. It is designed to allow command execution on endpoints.
+
+This document describes how commands are delivered to endpoints and how the server is notified of command results. This document does not cover who and how initiates command, and how results are interpreted.
+
+# Language
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+- **Command** is an action requested to be executed on a specific endpoint that may produce a result or fail.
+- **Command Invocation** is an act of a command entering the system and being scheduled for execution.
+- **Command Execution** is an act of transfering a command to endpoint for actual execution and waiting for execution results.
+
+# Requirements and constraints
+- Command execution is asynchronous in nature and may take time. Furthermore, the endpoint may reboot or client may reconnect as a part of the command execution sequence.
+- An endpoint may support multiple command types, thus it must tell them apart.
+- Command may success or fail. The caller must know outcomes of the command or the reason of a failure.
+- Commands are often not idempotent. That is, executing a single command twice may result in a different (possibly undesired) outcome.
+- An endpoint may be offline at the time of command invocation. The caller must be informed that the command execution did not succeed.
+
+# Use cases
+## UC1: Remote control
+An endpoint operates a device. Command Execution Protocol may be used to transfer commands to the endpoint and instruct it to change device behavior.
+
+## UC2: Expose local resources
```

Comment:
URL is not a requirement; it is one of the possible ways to access a resource.

And we do define resource path for addressing. You can think of `/<endpoint_token>/command/temperature` as a URL. Furthermore, these resource paths can be proxied via a REST, so there will be a real URL.